### PR TITLE
Implemented i18n on frontend

### LINKED ISSUE #246: Setup i18n for frontend
All the English strings and DateTime formatting should be put into a config file.

### PR DESCRIPTION
**Pull request by Trinovantes**
Closes #246

### CODE REVIEW

File: src-backbone/app/js/sanaApp.js [hunk 1]
```diff
@@ -19,44 +21,81 @@ module.exports = Marionette.Application.extend({
...
-        this._rootView.clearNotifications();
-    },
+        // Normalize path such that there's no leading slash
+        let urlPath = Backbone.history.location.pathname;
+        urlPath = (urlPath.charAt(0) === '/') ? urlPath.substr(1) : urlPath;
+        let pathParts = urlPath.split('/');
+
+        if (pathParts.length === 1 && pathParts[0].length === 0) {
+            // Index with no language
+            redirect(localeToLang(Config.DEFAULT_LOCALE));
```
Comment (Ommy) [inline src-backbone/app/js/sanaApp.js]: Why isn't this redirect with the `urlPath`?

File: src-backbone/app/js/templates/auth/loginView.hbs [hunk 2]
```diff
@@ -1,30 +1,30 @@
 <form>
     <div class="form-group">
-        <label for="username" class="sr-only">Username</label>
+        <label for="username" class="sr-only">{{ i18n "Username" }}</label>
```
Comment (Ommy) [inline src-backbone/app/js/templates/auth/loginView.hbs]: So will we have to write `i18n` for every element in a template?

File: src-backbone/app/js/sanaApp.js (hunk 1 above)
Comment (Trinovantes) [inline src-backbone/app/js/sanaApp.js]: `urlPath` is empty in this case

File: src-backbone/app/js/templates/auth/loginView.hbs (hunk 2 above)
Comment (Trinovantes) [inline src-backbone/app/js/templates/auth/loginView.hbs]: if you want the string translated

File: src-backbone/app/locales/en/translation.json [hunk 3]
```diff
@@ -0,0 +1,107 @@
+{
+
+    // Sign up
```
Comment (connorcimowsky) [inline src-backbone/app/locales/en/translation.json]: Are comments legal in a JSON file?

**Comment by connorcimowsky** [comment]
1 question, otherwise LGTM

File: src-backbone/app/locales/en/translation.json (hunk 3 above)
Comment (Trinovantes) [inline src-backbone/app/locales/en/translation.json]: Technically no. The preprocessor strips these comments out.

These translation files cannot be used by other apps unless they also strip the comment out first.

**Pull request merged**